samples/nrf_desktop/src/modules/usb_state.c

```diff
@@ -35,6 +35,32 @@ static int get_report(struct usb_setup_packet *setup, s32_t *len, u8_t **data)
 	return 0;
 }
 
+static int set_report(struct usb_setup_packet *setup, s32_t *len, u8_t **data)
+{
+	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE)) {
+		if ((*len) > 0) {
+			if ((*data)[0] != REPORT_ID_USER_CONFIG) {
+				LOG_WRN("Unsupported report ID");
+				return -ENOTSUP;
+			}
+
+			if ((*len) != (REPORT_SIZE_USER_CONFIG + 1)) {
+				LOG_WRN("Unsupported report size");
+				return -ENOTSUP;
+			}
+
+			struct config_event *event = new_config_event();
+
+			event->id = (*data)[1];
+			memcpy(event->data, &(data[0][2]), sizeof(event->data));
```

Comment:
`&((*data)[2])`

but I think it might be better idea to declare temporary variables at beginning of block e.g.:
```
__ASSERT...(len != NULL);
__ASSERT...((data != NULL) && (*data != NULL));
if (IS_ENABLED(x) && (*len > 0)) {
   size_t length = *len;
   const u8_t *buffer = *data;

   ...
}